### PR TITLE
Issue #2420 - fix 500 error on User Profile

### LINKED ISSUE #2420: User profile: user encounters 500 error when saving user profile information
### Current Behavior

Upon saving, user enters 500 error after updating user profile information.

image

### Expected Behavior

User can save user prolife information successfully. 

### Steps to Reproduce

On scenario1: User profile for current users ([as documented here](https://docs.google.com/document/d/1uzZy6a7u4uc9lLtfHfYpWQmGpmNFE_z7mE2GIDqhveY/edit))

Test 1: Follow steps 1-2 here, go to the user profile link in the upper right corner and delete a field. Click save and see this error 

Test 2: Follow steps 1-2 as a superuser who is “invited” go to the user profile link in the upper right corner and add to the “middle name” field.  
- For your superuser, you are not redirected to the finish setup page (at least for user Zander Adkinson). When trying to add those missing fields through your profile directly and clicking save, you get the same 500 error. For me it's this: django.urls.exceptions.NoReverseMatch: Reverse for '' not found. '' is not a valid view function or pattern name.
- After this error, if I click back and then try to fill in the field I deleted and save it, it then shows a 404 error page. Refreshing and then saving worked.

### Environment

_No response_

### Additional Context

_No response_

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by CocoByte**
## Ticket 2420

Resolves #2420

## Changes

- Added a check for form validation when setting up the user profile context
+ (BONUS) also changed a typo on the domain request intro (missing period)

## Code Review Verification Steps

code is deployed on [sandbox-nl](https://getgov-nl.app.cloud.gov/)

- navigate to user profile
- delete data in one of the required fields and try to save.

Verify that:
- [ ] instead of a 500 error, you now see the page refresh with error messages indicating that you are missing a required field

- start a new domain request
Verify that:
- [ ] the preamble now has a period where one was previously missing (image below shows missing period)
image

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by dave-kennedy-ecs** [comment]
After submitting all profile information on finish-profile-setup, I am not able to click the Logo and go back to home page, and I am not given a link to manage my domains
image

**Review by therealslimhsiehdy — COMMENTED** [review]
Ran into some issues, and here's some steps to reference in case for debugging:

This was me logging in as my superuser for the first time - your profile does show up, but I'm unable to click .gov admin to get back to the Domain Request page:
Screenshot 2024-07-03 at 4 47 06 PM

I manually changed the URL to `/` and then clicked on `your profile` -- no more 500 though 🥳 
Screenshot 2024-07-03 at 4 47 37 PM

Still on the `/` to `your profile` path -- my organization email disappeared:
Screenshot 2024-07-03 at 4 47 44 PM

Spelling error is fixed though and everything does display here, yay 🎆 
Screenshot 2024-07-03 at 4 49 54 PM

File: src/registrar/tests/test_views.py [hunk 1]
```diff
@@ -730,6 +731,8 @@ def test_new_user_with_profile_feature_on(self):
             self.assertContains(save_page, "Your profile has been updated.")
 
             # We need to assert that logo is not clickable and links to manage your domain are not present
+            # NOTE: "anage" is not a typo.  It is to accomodate the fact that the "m" is uppercase in one
```
Comment (dave-kennedy-ecs) [inline src/registrar/tests/test_views.py:734]: thanks for adding this

File: src/registrar/views/user_profile.py [hunk 2]
```diff
@@ -109,6 +108,8 @@ def form_invalid(self, form):
         """If the form is invalid, conditionally display an additional error."""
         if hasattr(self.user, "finished_setup") and not self.user.finished_setup:
             messages.error(self.request, "Before you can manage your domain, we need you to add contact information.")
+        form.initial["redirect"] = form.data.get("redirect")
```
Comment (dave-kennedy-ecs) [inline src/registrar/views/user_profile.py:111]: looks good.  remember to remove log before merging

**Review by dave-kennedy-ecs — APPROVED** [review]
LGTM

**Pull request merged**